### DISCUSSION TITLE
Why is my design scanning every other letter?

### DISCUSSION THREAD

**Discussion by aetkie** [UNC-Project-Open-AAC/OS-DPI#224] [Help]
Working on different scanning patterns with [this keyboard design](https://unc-project-open-aac.github.io/OS-DPI/#keyboard_predict_pattern_examples_1). I can't figure out why it is scanning every other letter instead of every letter?

https://github.com/UNC-Project-Open-AAC/OS-DPI/assets/113473933/9e84ab49-fe42-453d-9712-c38a9e2ad6e9

**Comment by gbishop** [comment]
I can't look now but I guess you have 2 methods firing

**Reply by aetkie** [reply]
I have only 2 switch active. Is it something in my spreasheet? 
[keyboard_predict_pattern_examples_1 (1).xlsx](https://github.com/UNC-Project-Open-AAC/OS-DPI/files/12686361/keyboard_predict_pattern_examples_1.1.xlsx)

**Comment by gbishop** [comment]
Export and send your current design

**Reply by aetkie** [reply]
[keyboard_predict_pattern_examples_1.osdpi.zip](https://github.com/UNC-Project-Open-AAC/OS-DPI/files/12686634/keyboard_predict_pattern_examples_1.osdpi.zip)

**Reply by gbishop** [reply]
It works as expected for me.

**Reply by aetkie** [reply]
How could that be? What's different between me in the design and you in the exported file?

**Comment by aetkie** [comment]
From: Gary Bishop ***@***.***>
Date: Thursday, September 21, 2023 at 11:36 AM
To: UNC-Project-Open-AAC/OS-DPI ***@***.***>
Cc: Etkie, Andrea Buonaiuto ***@***.***>, Author ***@***.***>
Subject: Re: [UNC-Project-Open-AAC/OS-DPI] Why is my design scanning every other letter? (Discussion #224)
Export and send your current design

**Comment by gbishop** [comment]
I'll have to look later

**Comment by gbishop** [comment]
Refresh your browser. Maybe that will do it

**Reply by aetkie** [reply]
That worked. But it's odd to me still that that worked. Because I had the design open days ago, and was still experiencing that issue. Today I opened it back up and still happened.

I dunno.. 

Thanks @gbishop !

**Reply by gbishop** [reply]
There could be an important clue here. You say you "opened it back up". Are you saying you *closed* the tab and used the File -> Open menu to re-open it in a new tab?

**Reply by aetkie** [reply]
Yes, that's exactly what I did.

**Reply by gbishop** [reply]
Then refreshing the browser should not have fixed it. Strange.

**Reply by gbishop** [reply]
Did you use Undo? I'm grasping at straws for what causes this.

**Comment by gbishop** [comment]
My current guess is that you must have used Undo at some point in your
experimentation. I suspect there is a lurking bug in it. But I'm just
guessing. I need a way to run a post mortem on bugs like this.

**Comment by gbishop** [comment]
I have pushed a change to production that I hope will prevent this from happening. Please let me know if you see it again.
